app_server/app/adapters/socket/Client.js

```diff
@@ -25,46 +27,70 @@ var Class = Client.prototype;
 
 Client.EVENT_COMMENT = Class.EVENT_COMMENT = 'comment';
 
-Class.getUsername = function getUsername() {
-  return this.credentials['username'];
+Class.getUserId = function getUserId() {
+  return this.credentials['userId'];
 };
 
 Class.getCredentials = function getCredentials() {
   return this.credentials;
 };
 
-Class.getSocket = function getSocket() {
-  return this.socket;
+Class.getSocketId = function getSocketId() {
+  return this.socket.id;
 };
 
-Class.getRoom = function getRoom() {
-  return this.room;
+/**
+ * Joins a socket.io room represented by a Room instance
+ * @param room {Room}
+ */
+Class.joinRoom = function joinRoom(room) {
+  this.socket.join(room.getName());
+  this.rooms[room.getName()] = room;
 };
 
-Class.joinRoom = function joinRoom(roomName) {
-  this.socket.join(roomName);
-  this.room = roomName;
+/**
+ * Leave a socket.io room represented by a Room instance
+ * @param room {Room}
+ */
+Class.leaveRoom = function leaveRoom(room) {
+  if (!(room.getName() in this.rooms)) {
+    let err = `Client ${this.getSocketId()} is not in ${room.getName()}`;
```

Comment:
Are we migrating to use `let` and `const` instead of `var` over the entire server codebase?
I'm not sure it's a good idea to mix `var` with `let` and `const`.